src/networking/SocketListener.java

```diff
@@ -0,0 +1,274 @@
+ * has the functionality of threads. This thread basically keeps listening on a port for the client requests, whenever
+ * there is a request it accepts(In a blocking manner) them and connects them to another socket which basically
+ * receives the message. After receiving the message from the client these messages are 
+ * pushed into either content module queue or processing module queue based on the identifier. 
+ * 
+ * @author Marella Shiva Sai Teja
+ */
+
+public class SocketListener implements Runnable {
+	
+	/**
+	 * Port number on which the socket will be listening for the client requests to connect with.
+	 */
+	private int port;
+
+	/**
+	 * socket object which keeps listening for the client requests.
+	 */
+	private ServerSocket serverSocket;
+
+	/**
+	 * Queue into which message from content module will be pushed
+	 */
+	private IQueue<IncomingPacket> contModuleQueue;
+
+	/**
+	 * Queue into which message from processing module will be pushed
+	 */
+	private IQueue<IncomingPacket> procModuleQueue;
+
+	/** 
+	 * logger object from the LoggerFactory to log messages
+	*/
+	ILogger logger = LoggerFactory.getLoggerInstance();
+
+	/**
+	 * 
+	 * This method is the constructor of the class which initializes the params
+	 * @param port
+	 * @param contModuleQueue
+	 * @param procModuleQueue
+	 * 
+	 * There won't be any return type as it is a constructor of the class
+	 */
+	public SocketListener(int port, IQueue<IncomingPacket> contModuleQueue, IQueue<IncomingPacket> procModuleQueue){
+		this.port = port;
+		this.contModuleQueue = contModuleQueue;
+		this.procModuleQueue = procModuleQueue;
+	}
+
+	/**
+	 * This method parses the identifier within packet received from client.
+	 * @param packet Data received from the client
+	 * @return Identifier
+	 */
+	public String getIdFromPacket(String packet){
+		/**
+		 * converts String to char[] array
+		 */
+		char[] chars = packet.toCharArray();
+		
+		/**
+		 * String Builder to build a string for the identifier
+		 */
+		StringBuilder idBuilder = new StringBuilder();
+		
+		/**
+		 * Iterates over char[] array and appends to the Identifier Builder
+		 */
+		for (char ch : chars) {
+			if(ch == ':'){
+				break;
+			}
+			idBuilder.append(ch);
+		}
+
+		/**
+		 * Converts the String Builder to String
+		 */
+		String id = idBuilder.toString();
+		return id;
+	}
+
+	/**
+	 * This method parses the message within the packet received from client.
+	 * @param packet Data received from the client
+	 * @return Message
+	 */
+	public String getMsgFromPacket(String packet){
+
+		char[] chars = packet.toCharArray();
+		
+		/**
+		 * String Builder to build required message
+		 */
+		StringBuilder msgBuilder = new StringBuilder();
+
+		/**
+		 * A Utility Boolean flag 
+		 */
+		boolean flag = false;
+		for (char ch : chars) {
+			if(flag){
+				msgBuilder.append(ch);
+			}
+			if(ch == ':'){
+				flag = true;
+			}
+		}
```

Comment:
This too.

Reply:
@convict-git modified the code using JSON library

Reply:
@convict-git  it is ready to be merge now I guess.